system/jlib/jsocket.cpp

```diff
@@ -2560,8 +2579,7 @@ ISocket* ISocket::multicast_connect(const SocketEndpoint &ep, unsigned _ttl)
 {
     Owned<CSocket> sock = new CSocket(ep,sm_multicast,NULL);
     sock->udpconnect();
-    u_char ttl = _ttl;
-    setsockopt(sock->OShandle(), IPPROTO_IP, IP_MULTICAST_TTL, (char *) &ttl, sizeof(ttl));
+    sock->set_ttl(_ttl);
```

Comment:
I did this check in set_ttl() but for clarity and saving a call I will add this check.